src/domain/data/index.ts

```diff
@@ -0,0 +1,44 @@
+export class DataProvider {
+  private readonly encoder = new TextEncoder();
+  constructor(
+    private readonly textLen: number,
+    private readonly patternLen: number
+  ) {}
+
+  public getData(): DataSet {
+    const text = this.getRandomString(this.textLen);
+    const subStringIndex = Math.round(
+      Math.random() * (text.length - this.patternLen)
+    );
+    const pattern = text.substring(
+      subStringIndex,
+      subStringIndex + this.patternLen
+    );
+
+    return new DataSet(
+      this.encoder.encode(text),
+      this.encoder.encode(pattern),
+      text.length
+    );
+  }
+
+  private getRandomString(length: number, randomString = ""): string {
+    randomString += Math.random().toString(20).substr(2, length);
+    if (randomString.length > length) {
+      return randomString.slice(0, length);
+    }
+    return this.getRandomString(length, randomString);
```

Comment:
Consider rewriting as an iteration (in the loop)

Reply:
Good point. At larger text lengths (>1000000) this method overflows because of recursion. I'll refactor this.

Reply:
Reimplemented `getRandomString`. Worked well even for 10^9 text length.